### ISSUE TITLE
Moving away from triangle fans makes the Unity plugin crash

### ISSUE THREAD

**Issue by toolness** [servo/pathfinder#178]
Ahoy! I was just updating the Unity plugin to the latest rev of Pathfinder in https://github.com/toolness/pathfinder-unity-fun/pull/2 and noticed that, as of acf666b70161228f139b8cc9c4e03a75a5c1101c, "Stop using triangle fans for DX12/Vulkan compatibility", the plugin can't render more than one frame before segfaulting.  Using the revision before it works fine, however.

Any idea what might be going on here, @pcwalton?

**Comment by pcwalton** [comment]
Not without more information. Do you have a stack trace?

**Comment by toolness** [comment]
Oops, sorry about that!  This is the exception:

```
Exception thrown at 0x00000000765479E0 (nvoglv64.dll) in unity-project.exe: 0xC0000005: Access violation reading location 0x0000000000000000. occurred
```

Here's the stack trace:

```
 	nvoglv64.dll!00000000765479e0()	Unknown
 	nvoglv64.dll!0000000075d549d7()	Unknown
>	GfxPluginPathfinder.dll!gl::DrawElementsInstanced(unsigned int mode, int count, unsigned int type_, core::ffi::c_void * indices, int instancecount) Line 1842	Unknown
 	GfxPluginPathfinder.dll!pathfinder_gl::{{impl}}::draw_elements_instanced(pathfinder_gl::GLDevice * self, pathfinder_gpu::Primitive primitive, unsigned int index_count, unsigned int instance_count, pathfinder_gpu::RenderState * render_state) Line 564	Unknown
 	GfxPluginPathfinder.dll!pathfinder_renderer::gpu::renderer::Renderer<pathfinder_gl::GLDevice>::draw_alpha_tiles<pathfinder_gl::GLDevice>(unsigned int self) Line 552	Unknown
 	GfxPluginPathfinder.dll!pathfinder_renderer::gpu::renderer::Renderer<pathfinder_gl::GLDevice>::render_command<pathfinder_gl::GLDevice>(pathfinder_renderer::gpu_data::RenderCommand * self) Line 251	Unknown
 	GfxPluginPathfinder.dll!pathfinder_renderer::concurrent::scene_proxy::SceneProxy::build_and_render<pathfinder_gl::GLDevice>(pathfinder_renderer::gpu::renderer::Renderer<pathfinder_gl::GLDevice> * self, pathfinder_renderer::options::RenderOptions renderer) Line 88	Unknown
 	GfxPluginPathfinder.dll!pathfinder_c_api_fun::render::Renderer::render(pathfinder_canvas::CanvasRenderingContext2D * self) Line 76	Unknown
 	GfxPluginPathfinder.dll!pathfinder_c_api_fun::PluginState::render() Line 140	Unknown
 	GfxPluginPathfinder.dll!pathfinder_c_api_fun::handle_render_event(int _event_id) Line 197	Unknown
 	UnityPlayer.dll!00007ffb0e6a2342()	Unknown
 	UnityPlayer.dll!00007ffb0dd05337()	Unknown
 	UnityPlayer.dll!00007ffb0dd09a8b()	Unknown
 	UnityPlayer.dll!00007ffb0dd09b85()	Unknown
 	UnityPlayer.dll!00007ffb0e4d8394()	Unknown
 	[External Code]	
```

And here are the locals at the point that `gl::DrawElementsInstanced()` explodes:

```
		count	6	int
+		indices	0x0000000000000000 {???}	core::ffi::c_void *
		instancecount	118	int
		mode	4	unsigned int
		type_	5125	unsigned int
```

Also, this crash seems to be deterministic in that I can run it over and over and get the same stack trace with the same locals.

**Comment by pcwalton** [comment]
Do you have instructions for building the plugin? It'll be hard to debug that without building it locally…

**Comment by toolness** [comment]
Yep, I've tried documenting it as best I can here: https://github.com/toolness/pathfinder-unity-fun#quick-start

Let me know if there's anything that seems unclear!

**Comment by toolness** [comment]
Ok, I temporarily fixed this in #197 but then @pcwalton's merging of the metal branch into master obviated the need for that!  So this bug is smooshed and I will close it.
